tests/pre_merge/datasets/test_dataset.py

```diff
@@ -1,243 +1,74 @@
-
-    def test_to_numpy_channel_order(self, data_sample):
-        """ToNumpy() should return a numpy array of order [HxWxC]"""
-        array = ToNumpy()(data_sample["image"])
-        assert len(array.shape) == 3 and array.shape[-1] == 3
-
-    def test_one_channel_images(self, data_sample):
-        """One channel tensor should be converted to HxW np array."""
-        data = data_sample["image"][:, 0, :, :].unsqueeze(0)
-        array = ToNumpy()(data)
-        assert len(array.shape) == 2
-
-    def test_representation(self):
-        """Test ToNumpy() representation should return string `ToNumpy()`"""
-        assert str(ToNumpy()) == "ToNumpy()"
-
-
-class TestConfigToDataModule:
-    """Tests that check if the dataset parameters in the config achieve the desired effect."""
-
-    @pytest.mark.parametrize(
-        ["input_size", "effective_image_size"],
-        [
-            (512, (512, 512)),
-            ((245, 276), (245, 276)),
-            ((263, 134), (263, 134)),
-            ((267, 267), (267, 267)),
-        ],
     )
-    @TestDataset(num_train=20, num_test=10)
-    def test_image_size(self, input_size, effective_image_size, category="shapes", path=None):
-        """Test if the image size parameter works as expected."""
-        configurable_parameters = get_test_configurable_parameters(dataset_path=path, model_name="stfpm")
-        configurable_parameters.dataset.category = category
-        configurable_parameters.dataset.image_size = input_size
-        configurable_parameters = update_input_size_config(configurable_parameters)
-
-        data_module = get_datamodule(configurable_parameters)
-        data_module.setup()
-        assert iter(data_module.train_dataloader()).__next__()["image"].shape[-2:] == effective_image_size
+    dataset.setup()
+
+    return dataset
+
+
+class TestAnomalibDataset:
+    def test_subsample(self, folder_dataset):
+        """Test the subsample functionality."""
+
+        sample_size = int(0.5 * len(folder_dataset))
+        indices = random.sample(range(len(folder_dataset)), sample_size)
+        subset = folder_dataset.subsample(indices)
+
+        # check if the dataset has been subsampled to correct size
+        assert len(subset) == sample_size
+        # check if index has been reset
+        assert subset.samples.index.start == 0
+        assert subset.samples.index.stop == sample_size
+
+    def test_random_split(self, folder_dataset):
+        """Test the random subset splitting."""
+
+        # subset splitting
+        subsets = random_split(folder_dataset, [0.4, 0.35, 0.25])
+        # check if subset splitting has been performed correctly
+        assert len(subsets) == 3
+        # reconstruct the original dataset by concatenating the subsets
+        reconstructed_dataset = concatenate_datasets(subsets)
+        # check if reconstructed dataset is equal to original dataset
+        assert folder_dataset.samples.equals(reconstructed_dataset.samples)
```

Comment:
picky comment: not a great pratice to test your lib by using another util from your own lib 

for instance

```suggestion
        folder_dataset_indices = set(folder_dataset.samples.index)
        subsets_indices = set.union(*[set(ss.samples.index) for ss in subsets])
        assert folder_dataset_indices == subsets_indices, f"diff {folder_dataset_indices.symmetric_difference(subsets_indices)}"
```

i also added this message `f"diff {folder_dataset_indices.symmetric_difference(subsets_indices)}"` to make it readily debuggable as a suggesttion but maybe it's discussible wether this should happen in a test suite (i believe so because it makes easier to dig into errors but the code will grow a bit if this is done everywhere)